### ISSUE TITLE
Provide Quotas

### ISSUE THREAD

**Issue by eloo** [promregator/promregator#90]
Hi,
i would like to request a new feature which provides the org/space quotas.
This would be very useful to have a whole overview of the monitored spaces/orgs and prevent missing deployments or scaling because the lack of resources. 

Also it looks like one call per org is enough to fetch this informations of the usage and two calls to get the quota itself.
Follo
```
cf curl /v2/organizations/<org-guid>/summary    // get usage
cf quotas    // get available quotas
cf <org-name>   // get the assigned quota
```

@eaglerainbow What do you think?

**Comment by eaglerainbow** [comment]
Mmmh... general good idea to have it.
I get your point and I totally understand what you want to achieve.
In general, I still have one doubt with it, because such a feature would change the market positioning of Promregator in general: So far Promregator "only" is a catalyst for establishing the communication between Prometheus and the a-little-strange exposure of the metrics API for applications running on Cloud Foundry. The metrics, which we collect along the way, is more like a by-catch: We have to know the information anyway - and it might be helpful to convey their values along to Prometheus such that the administrator is enabled to see that Promregator's data transfer is stable, well-performing, etc. Retrieving them is not the primary focus of Promregator. 
Fetching necessary metrics which are available from the CF API should be more a matter of an exporter (such as https://github.com/bosh-prometheus/cf_exporter, which, however, is on BOSH level and thus not usable for application developers). Though Promregator often behaves similar to an exporter, it isn't one (that's also why it is listed under "Others" only at https://prometheus.io/docs/operating/integrations/ ). The feature request you make is clearly asking for something that's sounds to me being a task for an exporter. Integrating that into Promregator to me feels quite like violating the [DOTADIW principle](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well) (and I am quite a fan of that).

Perhaps this is a second "product" in the yet-very-small Promregator family? Or something like a "plugin"? I don't know...

So, in short: Point understood and taken, but the appropriate positioning/design is yet very unclear to me.
Just my $0.02....

**Comment by eaglerainbow** [comment]
One more thing which comes to my mind reading this again a couple of days later: 

In #83 we saw that the environment we have (both on the client side - and perhaps also on the CF Cloud Controller side) doesn't like to be flooded with requests. That is why Promregator heavily relies on caching that data. Having such kind of metrics mentioned in this issue here on my mind, you do not want to have them cached (that totally jeopardizes their intention). Thus, if your Prometheus has a (global) scraping interval of 15s, this means that you would request the quota stuff latest every 15 seconds from the CFCC. Calculating quota usage, on the other, can be quite costly for the CF Cloud Controller (these are transient/derived values - and not persisted master data data, which is mainly what Promregator polls today). For sure nothing that you would compute just to "have it on stock".

I don't know how happy operators of the CFCC would be to have several CF exporters polling their expensive metrics all the time. :fearful:
